File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        char[] chars = s.toCharArray();
        int m = 0;
        int count = 0;
        
        for (int i=chars.length-1; i > -1; i--){
            if (chars[i] == ')'){
                count++;
                m = Math.max(m, count);
            }
            else if (chars[i] == '('){
                count--;
                m = Math.max(m, count);
            }
        }
        return m;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        if len(s) == 1:
            return 0
        ls = list(s)[::-1]
        m = 0
        count = 0
        
        for i in range(len(ls)):
            if ls[i] == ")":
                count += 1
                m = max(m, count)
            elif ls[i] == "(":
                count -= 1
                m = max(m, count)
            else:
                continue
                
        return m